packages/analytics-browser/src/plugins/file-download-tracking.ts

```diff
@@ -0,0 +1,50 @@
+import { BrowserClient, PluginType, Event, EnrichmentPlugin } from '@amplitude/analytics-types';
+import { BrowserConfig } from '../config';
+
+const FILE_DOWNLOAD_EVENT = 'file_download';
+
+export const fileDownloadTracking = (): EnrichmentPlugin => {
+  const name = 'fileDownloadTracking';
+  const type = PluginType.ENRICHMENT;
+  const setup = async (config: BrowserConfig, amplitude?: BrowserClient) => {
+    /* istanbul ignore if */
+    if (!amplitude) {
+      // TODO: Add required minimum version of @amplitude/analytics-browser
+      config.loggerProvider.warn(
+        'File download tracking requires a later version of @amplitude/analytics-browser. File download events are not tracked.',
+      );
+      return;
+    }
+
+    const ext =
+      /\.(pdf|xlsx?|docx?|txt|rtf|csv|exe|key|pp(s|t|tx)|7z|pkg|rar|gz|zip|avi|mov|mp4|mpe?g|wmv|midi?|mp3|wav|wma)$/;
+    const links = Array.from(document.getElementsByTagName('a'));
```

Comment:
I think this only supports pre-rendered static pages. Do we want to support dynamic pages like SPA? When the `setup` is invoked, if the `a` tag is not rendered on the page yet, or user interaction will create a `a` tag on the fly (like navbar, popup window), then this new `a` tag will be missed. If we want to support this, we can use [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe) to watch `document` sub tree addition and filter to `a` node. There might have other solutions too.

Reply:
Good call out! I can confirm we want to handle this case as well. I'll look into how to best handle this scenario.